emcc.py

```diff
@@ -1242,6 +1242,12 @@ def check(input_file):
       if shared.Settings.WASM:
         shared.Settings.BINARYEN = 1 # these are synonyms
 
+        # When only targeting wasm, the .asm.js file is not executable, so is treated as an intermediate build file that can be cleaned up.
+        if shared.Building.is_wasm_only():
+          asm_target = asm_target.replace('.asm.js', '.temp.asm.js')
```

Comment:
Oh ok. What we want is a mechanism for what people should do when they want to co-deploy both asm.js and wasm builds which both have maximal performance (i.e. resort to building twice).

In either case, I think it's best not to generate out temporary files that potentially can collide with other build artifacts that might pre-exist in the build output directory, because that can be surprising to users.